kernel/src/device/tdxguest/mod.rs

```diff
@@ -84,22 +130,10 @@ impl FileIo for TdxGuest {
     }
 }
 
-fn handle_get_report(arg: usize) -> Result<i32> {
-    const SHARED_BIT: u8 = 51;
-    const SHARED_MASK: u64 = 1u64 << SHARED_BIT;
-    let current_task = ostd::task::Task::current().unwrap();
-    let user_space = CurrentUserSpace::new(current_task.as_thread_local().unwrap());
-    let user_request: TdxReportRequest = user_space.read_val(arg)?;
-
+fn tdx_get_report(inblob: &[u8]) -> Result<Box<[u8]>> {
     let segment = FrameAllocOptions::new().alloc_segment(2).unwrap();
     let dma_coherent = DmaCoherent::map(segment.into(), false).unwrap();
-    dma_coherent
-        .write_bytes(0, &user_request.report_data)
-        .unwrap();
-    // 1024-byte alignment.
-    dma_coherent
-        .write_bytes(1024, &user_request.tdx_report)
-        .unwrap();
+    dma_coherent.write_bytes(0, &inblob).unwrap();
```

Comment:
Why this `write_bytes` is guaranteed to succeed? One potential reason for failure is that the length of `inblob` may be greater than the available space of `dma_coherent`. See [its API doc](https://asterinas.github.io/api-docs-nightly/ostd/mm/io/trait.VmIo.html#method.write_bytes):

> Writes a specified number of bytes from a given buffer at a specified offset.
>
> No short writes
> Similar to write.

In the original code, this possibility can be ruled out because the input data structure, `TdxReportRequest`, resides just a few lines above. We statically know that the size of `TdxReportRequest` does not exceed 2 pages. But I can't do the same reasoning in the new version.